File: src/workflows/workflowRuntimeEngine.ts
/**
 * Converted from Python: core/workflows/workflow_runtime_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildWorkflowRuntimeContext(url: any = "", runtime: any = "browser", sources: any = null): any {
  sources = py.or2(sources, () => ({}));
  return {"url": url, "primary_runtime": runtime, "sources": py.sorted(py.keys(sources)), "bounded": true};
}

File: lib/src/workflows/workflow_runtime_engine.dart
// Port of core/workflows/workflow_runtime_engine.py

Map<String, dynamic> buildWorkflowRuntimeContext({
  String url = '',
  String runtime = 'browser',
  Map<String, dynamic>? sources,
}) {
  final Map<String, dynamic> src = sources ?? <String, dynamic>{};
  final List<String> sortedKeys = src.keys.toList()..sort();

  return <String, dynamic>{
    'url': url,
    'primary_runtime': runtime,
    'sources': sortedKeys,
    'bounded': true,
  };
}
